gpumon/gpumon_client.ml

```diff
@@ -0,0 +1,31 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Gpumon_interface
+open Xcp_client
+
+let json_url () = "file:" ^ json_path
```

Comment:
First of all, I don't see `json_url` used anywhere, so it looks like dead code that should just be removed.

However, the same question applies to `xml_url` below. The place where this is used is in the function `http_rpc` in lib/xcp_client.ml, which passes it to `Uri.of_string`, which is provided by a library. I assume that this is the format that is accepted by this library, because indeed we use this format everywhere. For this PR, IMO, we should just comply with the other interfaces in xcp-idl, which is what the current patch does.